src/lib.rs

```diff
@@ -67,12 +67,19 @@
 //!
 //! Have a look at the examples to learn more about device and buffer management.
 
-#[cfg(feature = "v4l-sys")]
+#[cfg(all(feature = "libv4l", feature = "v4l2"))]
+compile_error!("You may not enable both `v4l-sys` and `v4l2-sys` features at the same time. Try disabling one of them. If you only specified one feature, you may wish to add `default-features = false` as a dependency key.");
+
+#[cfg(all(feature = "libv4l", not(feature = "v4l2")))]
 pub use v4l_sys;
 
-#[cfg(feature = "v4l2-sys")]
+#[cfg(all(feature = "v4l2", not(feature = "libv4l")))]
 pub use v4l2_sys as v4l_sys;
 
+// calms down rust-analyzer when `rust-analyzer.cargo.features = "all"`
```

Comment:
```suggestion
// calms down rust-analyzer when `rust-analyzer.cargo.features = "all"`, even though the compile_error!() above prevents this from being a valid, compilable configuration.
```

Perhaps move this to sit right below that line?

Reply:
Sounds good!

```suggestion
// calms down rust-analyzer when `rust-analyzer.cargo.features = "all"`,
// though the `compile_error!()` above prevents this from being a valid,
// compilable configuration.
```